### PR TITLE
Bug/#38

### PR DESCRIPTION
**Pull request by pedrohh**
Added better type check using `columnType.valueFromDB()` method and tests to validate implementation.

### CODE REVIEW

File: src/main/kotlin/test/ktortemplate/core/utils/pagination/TableExtensions.kt [hunk 1]
```diff
@@ -1,62 +1,72 @@
...
  */
 fun Table.createFilters(filters: List<FilterField>): List<Op<Boolean>> {
     return filters.map { filterField ->
-        val column = this.columns.single { it.name == filterField.field }
-        when (column.columnType) {
-            is LongColumnType, is AutoIncColumnType, is EntityIDColumnType<*> -> {
-                column as Column<Long>
-                filterField.values.map { value -> Op.build { column.eq(value.toLong()) } }.compoundOr()
+        val column: Column<*> = this.columns.single { it.name == filterField.field }
+        val valueFromDB = column.columnType.valueFromDB(filterField.values.first()).let {
```
Comment (pedrohh) [inline src/main/kotlin/test/ktortemplate/core/utils/pagination/TableExtensions.kt:39]: Using `columnType.valueFromDB()` as helper to get the correct column data type. If it is an EntityId, e.g. Entity<String>, we extract the value from it.

File: src/main/kotlin/test/ktortemplate/core/utils/pagination/TableExtensions.kt [hunk 2]
```diff
@@ -1,62 +1,72 @@
...
+            when (it) {
+                is EntityID<*> -> it.value
+                else -> it
+            }
+        }
+
+        // Support to more types can be added as needed bellow
+        when (valueFromDB) {
+            is Long -> column.asType<Long>().let {
```
Comment (pedrohh) [inline src/main/kotlin/test/ktortemplate/core/utils/pagination/TableExtensions.kt:48]: For each type we can safely convert it from **Column<*>** to **Column<Long>** using the extension method `asType()`

File: src/test/kotlin/test/ktortemplate/core/utils/pagination/TestTableExtensionsFilters.kt [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+    @AfterEach
+    fun cleanDatabase() {
+        dbc.query { TestTable.deleteAll() }
+    }
+
+    @Nested
+    inner class TestFilterId {
+        @Test
+        fun `String primary key type should be supported`() = testAppWithConfig {
```
Comment (felix19350) [inline src/test/kotlin/test/ktortemplate/core/utils/pagination/TestTableExtensionsFilters.kt:63]: For this particular test, wouldn't it be simpler to start the koin context and test the database access stuff without the overhead of the ktor application? Same as we did on `TestCarService.kt`

**Review by felix19350 — COMMENTED** [review]
I think we can further simplify the tests. For this case, unless I am missing something, we don't need to test in the context of the full ktor application.

File: src/test/kotlin/test/ktortemplate/core/utils/pagination/TestTableExtensionsFilters.kt (hunk 3 above)
Comment (pedrohh) [reply src/test/kotlin/test/ktortemplate/core/utils/pagination/TestTableExtensionsFilters.kt:63]: Yep thanks 👍

**Review by felix19350 — APPROVED** [review]

**Pull request merged**